central/processlisteningonport/datastore/pruning.go

```diff
@@ -31,4 +31,61 @@ const (
 	// if the serialized data has process information. PLOPs without process information are then deleted.
 	getPotentiallyOrphanedPLOPs = `SELECT plop.serialized FROM listening_endpoints plop where NOT EXISTS
 			(select 1 FROM process_indicators proc where plop.processindicatorid = proc.id)`
+
+	//// Finds PLOPs without poduids and deletes them. This is done in batches.
+	//deletePLOPsWithoutPoduid = `WITH temp AS
+	//	(SELECT id, poduid FROM listening_endpoints LIMIT %d OFFSET %d)
+	//	DELETE FROM listening_endpoints WHERE id IN (SELECT id FROM temp where poduid is null)`
+
+///////////////////////////////
+
+	// Finds PLOPs without poduids and deletes them. This is done in batches.
+	deletePLOPsWithoutPoduid = `WITH temp AS
```

Comment:
Why do we need to use the `offset` here?  Couldn't we just use `limit` delete that batch use `limit` again because the first cut of rows are now gone?

Reply:
I don't think so. If there are many rows with null poduids, that query will run many times, each time taking longer to run as it starts from the beginning and scans further, until it reaches the 10 second time limit. If the table is very large and has no null poduids the query will take longer than the 10 second time limit to execute.